side-nav: new variants for nested items (#1702)
agriculturegovau/agds-next: packages/react/src/side-nav/docs/overview.mdx

<DoHeading />

- include a concise label
- indent each sub-level

<DontHeading />

> - create more than 3 sub-levels

Reviewer: The original was talking about all levels right?
```suggestion
- create more than 2 sub-levels
```

- place on the right-hand side of content
- keep all lists with sub-levels open when there are many items.

## Sub-level items

To help users comprehend the side navigation, choose a strategy that is appropriate for the number of items in the list.

### Visible when active

> When the list of items is long, prefer showing the sub-level lists when the item is active. The item is considered active after it is clicked, as the user will be navigated to this page. This is the default behaviour so either setting `subLevelVisible="whenActive"` or omitting this prop will set this strategy.

Reviewer: Perhaps something like this?
```suggestion
When the list of items is long, prefer showing any sub-level lists when either navigating to the parent item or one of the sub-level items. Since this is the default behaviour, either setting `subLevelVisible="whenActive"` or omitting this prop will use this strategy.
```

	titleLink="#"
	collapseTitle="In this section"
	activePath="#in-detail/record-keeping"
	items={[
		{
			href: '#welcome',
			label: 'Welcome',
		},
		{
			href: '#do-you-need-to-lodge-a-tax-return',
			label: 'Do you need to lodge a tax return?',
		},
		{
			href: '#lodge-online',
			label: 'Lodge online',
			items: [
				{
					href: '#lodge-online/pre-filling',
					label: 'Pre-filling your online tax return',
				},
			],
		},
		{
			href: '#whats-new-for-individuals',
			label: 'What’s new for individuals',
		},
		{
			href: '#why-you-may-receive-a-tax-bill',
			label: 'Why you may receive a tax bill',
		},
		{
			href: '#in-detail',
			label: 'In detail',
			items: [
				{
					href: '#in-detail/record-keeping',
					label: 'Record keeping',
					items: [
						{
							href: '#in-detail/record-keeping/tax',
							label: 'Keeping your tax records',
						},
						{
							href: '#in-detail/record-keeping/incorrect-amounts',
							label: 'Incorrect amounts',
						},
					],
				},
				{
					href: '#tax-receipt',
					label: 'Tax receipt',
				},
				{
					href: '#pre-fill-availability',
					label: 'Pre-fill availability',
				},
			],
		},
	]}

> 	subLevelVisible="whenActive"

Reviewer: I know this follows our standard pattern on new props at the end, but I wonder if moving this above `items` will help users understand (since I was also confused that first time)

<DontHeading />

- create more than 3 levels of nesting
- place on the right-hand side of content

> - keep all nested lists open when there are many items.

Reviewer: What does this mean exactly? I'm confused

Author: Essentially don't use `alwaysOpen` on long lists of nav items, that's what `openOnNav` is for. Open to new wording.

Reviewer: I see. Perhaps we need a Do of use the "openOnNav" as the primary method? And Don't use "alwaysOpen" when there are many navigation items with many sub items?

Author: Yeah, that's a more complete approach, I like that.

How about this:

```md
DO:
- use `openOnNav` as a primary variant.

DON'T:
- use `alwaysOpen` when you have a long list of navigation items.
```

Reviewer: primary variant doesn't seem to mean much here? I know I used primary method above, and method does make it a big clearer, but still not clear enough imo. Perhaps it should be something along the lines of "use `openOnNav` unless you have very good reason not to do so"?

Also, I think we need guidance on when we do the click into a nav with sub options and only show those options like the current export service website. Is that still a thing?

Author: - [ ] Get designers' input on wording

## Nested items

To help users comprehend the side navigation, choose a variant that is appropriate for the number of items in the list.

### Always open

> When you have a small number of nested items, it's easier for users to see all items at once. Set `nestedItemsVariant="always-open"` to achieve this LanguageVariant.

Reviewer: ```suggestion
When you have a small number of nested items, it’s easier for users to see all items at once. Set `nestedItemsVariant="always-open"` to achieve this variant.
```
